src/components/PopoverWithMeasuredContent.js

```diff
@@ -0,0 +1,146 @@
+    }
+
+    /**
+     * Measure the size of the popover's content.
+     *
+     * @param {Object} nativeEvent
+     */
+    measurePopover({nativeEvent}) {
+        this.popoverWidth = nativeEvent.layout.width;
+        this.popoverHeight = nativeEvent.layout.height;
+        this.setState({isContentMeasured: true});
+    }
+
+    /**
+     * Calculate the adjusted position of the popover.
+     *
+     * @returns {Object}
+     */
+    calculateAdjustedAnchorPosition() {
+        let horizontalConstraint;
+        switch (this.props.anchorOrigin.horizontal) {
+            case CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.RIGHT:
+                horizontalConstraint = {left: this.props.anchorPosition.horizontal - this.popoverWidth};
+                break;
+            case CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.CENTER:
+                horizontalConstraint = {
+                    left: Math.floor(this.props.anchorPosition.horizontal - (this.popoverWidth / 2)),
+                };
+                break;
+            case CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT:
+            default:
+                horizontalConstraint = {left: this.props.anchorPosition.horizontal};
+        }
+
+        let verticalConstraint;
+        switch (this.props.anchorOrigin.vertical) {
+            case CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM:
+                verticalConstraint = {top: this.props.anchorPosition.vertical - this.popoverHeight};
+                break;
+            case CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.CENTER:
+                verticalConstraint = {
+                    top: Math.floor(this.props.anchorPosition.vertical - (this.popoverHeight / 2)),
+                };
+                break;
+            case CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.TOP:
+            default:
+                verticalConstraint = {top: this.props.anchorPosition.vertical};
+        }
+
+        return {
+            ...horizontalConstraint,
+            ...verticalConstraint,
+        };
+    }
+
+    render() {
+        return this.state.isContentMeasured
+            ? (
+                <Popover
+                    // eslint-disable-next-line react/jsx-props-no-spreading
+                    {...this.props}
+                    anchorPosition={this.calculateAdjustedAnchorPosition()}
+                >
+                    {this.props.children}
+                </Popover>
+            ) : (
+
+                /*
+                    This is an invisible view used to measure the size of the popover,
+                    before it ever needs to be displayed.
+                    We do this because we need to know its dimensions in order to correctly animate the popover,
+                    but we can't measure its dimensions without first animating it.
+                */
+                <View style={styles.invisible} onLayout={this.measurePopover}>
+                    {(this.props.measureContent || (() => this.props.children))()}
```

Comment:
I think I've asked this before but why are we doing this check for either `measureContent` or `this.props.children`? Seems like there are no cases where we will want to measure the children.

Reply:
> Seems like there are no cases where we will want to measure the children.

I was thinking this would be the default behavior of this component – the only reason we need to have a `measureContent` prop is because this component can't control if its children have styles that hide themselves. I think we can simplify this and just make the `measureContent` prop required.